### ISSUE TITLE
A bit confused by instructions and operation

### ISSUE THREAD

**Issue by romanovzky** [Chadsr/NordVPN-NetworkManager#69]
Hi there!
Thank you a lot for this tool, it is great to have this functionality on Arch Linux.
I'm just writing as I can't seem to understand how to best utilise this tool, and the instructions fall a bit short for my needs.
For example, imagine that I want to sync the best servers, I would run
```
$ sudo nordnm s -u
```
However, if I am **not already connected to a manually configured** NVPN server, the tool does not provide an updated server list. I get the output
```
[INFO] [nordnm.nordnm]: Downloading latest NordVPN OpenVPN configuration files to '/home/myusername/.nordnm/configs/'.
[INFO] [nordnm.nordnm]: Checking for new connections to import...
[INFO] [nordnm.nordnm]: Benchmarking servers...
[INFO] [nordnm.nordnm]: Benchmarking complete. Took 2.79 seconds.
[INFO] [nordnm.nordnm]: Removing all active connections...
[INFO] [nordnm.nordnm]: Adding new connections...
[INFO] [nordnm.nordnm]: No new connections added.
[INFO] [nordnm.networkmanager]: DNS leak protection enabled.
```
and **all nvpn connections auto-generated by the tool are removed**, without adding new ones.

If, on the other hand, I perform the same command while **connected to a manually configured** NVPVN server, the tool syncs and generates new connections
```
[INFO] [nordnm.nordnm]: Downloading latest NordVPN OpenVPN configuration files to '/home/myusername/.nordnm/configs/'.
[INFO] [nordnm.nordnm]: Configuration files already up-to-date.
[INFO] [nordnm.nordnm]: Checking for new connections to import...
[INFO] [nordnm.nordnm]: Benchmarking servers...
[INFO] [nordnm.nordnm]: Benchmarking complete. Took 1.93 seconds.
[INFO] [nordnm.nordnm]: No active connections to remove.
[INFO] [nordnm.nordnm]: Adding new connections...
[INFO] [nordnm.nordnm]: 5 new connections added.
[INFO] [nordnm.networkmanager]: DNS leak protection enabled.

```
which I can connect using the network manager, as expected.

Lastely, if I am **connected to an automatically generated** NVPN server, the tool identifies this, removes the added servers, but does not generate new ones
```
[INFO] [nordnm.nordnm]: Downloading latest NordVPN OpenVPN configuration files to '/home/myusername/.nordnm/configs/'.
[INFO] [nordnm.nordnm]: Configuration files already up-to-date.
[INFO] [nordnm.nordnm]: Checking for new connections to import...
[WARNING] [nordnm.nordnm]: Active VPN(s) disabled for accurate benchmarking. Your connection is not secure until these are re-enabled.
[INFO] [nordnm.nordnm]: Benchmarking servers...
[INFO] [nordnm.nordnm]: Benchmarking complete. Took 2.85 seconds.
[INFO] [nordnm.nordnm]: Removing all active connections...
[INFO] [nordnm.nordnm]: Adding new connections...
[INFO] [nordnm.nordnm]: No new connections added.
[INFO] [nordnm.networkmanager]: DNS leak protection enabled.
```

Long story short:

**I can only use the tool to generate new servers in my network manager if I am already connected to a manyally configured NVPN connection. All other attemps seem to delete any generated list of servers.**

Am I missing something or have I found a bug?

Cheers!

**Comment by Chadsr** [comment]
Hi and thanks for the detailed report.

This sounds very similar to an issue another user had (#48) some months ago. In that particular case, we discovered their network/ISP was blocking ICMP (pinging), which the benchmarking relies on heavily. 

If a server shows packet loss of > 5%, it is ignored... which can be problematic and not very intuitive if all servers are ignored.

To see if this is the same problem, could you run `ping -c 5 -n -i 0.2 -W 1 155.94.242.4` and tell me if the output shows significant packet loss, or just fails?

Regarding syncing when connected to an automatically generated connection (the last section). The active VPN is disconnected to improve the accuracy of the bencmkarking (so latency tests are from your actual location). This can however be disabled like so: `sudo nordnm sync --preserve-vpn` or `sudo nordnm s -p` for short.

**Comment by romanovzky** [comment]
Hi there,
Thanks for your quick reply.
The ping seems fine
```
$ ping -c 5 -n -i 0.2 -W 1 155.94.242.4
PING 155.94.242.4 (155.94.242.4) 56(84) bytes of data.
64 bytes from 155.94.242.4: icmp_seq=1 ttl=47 time=124 ms
64 bytes from 155.94.242.4: icmp_seq=2 ttl=47 time=127 ms
64 bytes from 155.94.242.4: icmp_seq=3 ttl=47 time=126 ms
64 bytes from 155.94.242.4: icmp_seq=4 ttl=47 time=126 ms
64 bytes from 155.94.242.4: icmp_seq=5 ttl=47 time=125 ms

--- 155.94.242.4 ping statistics ---
5 packets transmitted, 5 received, 0% packet loss, time 801ms
rtt min/avg/max/mdev = 124.325/126.245/127.316/1.140 ms
```

And don't worry about the ``--preserve-vpn``, I was just trying to provide a complete picture of the behaviour. I totally understand the reason it'd disconnect, and indeed what I want is to be able to get the list of servers before connecting to a VPN server so I can have the least latency possible.

**Comment by Chadsr** [comment]
Well that is odd. The ping results are with all VPNs disabled, correct?

Some ISPs have been know to block the nordvpn API, which is needed to retrieve server information. If that was the case here though, nordnm shouldn't even make it to benchmarking without throwing a fatal error.

Would you mind sharing your settings configuration (`~/.nordnm/settings.conf`) so I can attempt to replicate this problem?

I will make another note to improve the verbosity of benchmarking, so it's a little more obvious what's happening when the benchmarking process returns zero suitable servers.

**Comment by romanovzky** [comment]
Yape, initially I made the mistake of running whilst on a vpn, then I turned it off and run it to get the same result.
Also, I should add I have the same behaviour on both my Arch machines, which made me feel the problem was on how I was using the program...

```
$ less .nordnm/settings.conf
[Countries]
# simply write country codes separated by spaces e.g. country-blacklist = gb us
country-blacklist = 

# same as above. if this is non-empty, the blacklist is ignored
country-whitelist = GB

[Categories]
anti-ddos = true
dedicated-ip-servers = true
double-vpn = true
onion-over-vpn = true
p2p = true
standard-vpn-servers = true

[Protocols]
tcp = true
udp = true

[Benchmarking]
ping-attempts = 5
```

**Comment by Chadsr** [comment]
I did a quick test on Arch (which is what I develop this tool on) with your settings and didn't encounter any issues. It was an extremely quick test however.

The fact that this behaviour stops when benchmarking over a VPN connection, makes me believe that the problem has its roots in a networking issue. Just not one as simple as blocked ICMP.

My workload on other projects is pretty high for the next week, but I will work on improving the benchmarking output when I have the time. Hopefully some more verbosity helps us determine the root of the problem. If you find any more information in the meantime, please share it.

Thanks!

**Comment by romanovzky** [comment]
Cheers for looking into this, I look forward to see those changes so that we can assess what is happening.

I am not very knowledgeable about networking, so I don't know how to further test the situation. But could this be something with my router? All my attempts were with computers in my home network...

**Comment by Chadsr** [comment]
nordnm does make a LOT of simultaneous ping requests. It is possible that your router is either dropping ICMP packets when overwhelmed, or that it has a firewall preventing ping spam. If you have a firewall and see a setting mentioning ICMP flooding, try disabling it. If there's no ICMP specific setting try temporarily disabling your firewall and checking if that makes a difference.

It's a hard balance between making nordnm fast and not being labeled as a DoS bot :).

**Comment by peteruithoven** [comment]
@romanovzky maybe you could try it on a different location / network? Maybe try creating a hotspot on your phone and connecting through that.

**Comment by romanovzky** [comment]
Hi guys, problem solved. It was indeed a firewall IP flood detection system that was preventing the sync. Thank you both for your help, and maybe include in the manual/instructions a warning about this possibility.
Cheers

**Comment by Chadsr** [comment]
Glad to hear you found a solution!

I have made a note of all of your feedback and will aim to incorporate it as soon as I have some time. In the meantime, I hope the tool proves useful to you!
